ha/util/consul_kv_store.py

```diff
@@ -103,10 +116,20 @@ def key_exists(self, key: str):
         Return:
             bool: True if key exists else False.
         """
-        _, data = self._consul.kv.get(self._prepare_key(key), recurse=True)
-        if data is None:
-            return False
-        return True
+        k = self._prepare_key(key)
+
+        # First find in cache if batch put is enabled
+        Log.debug(f"[consul-op] is key exist in cache: {k}")
+        if self._enable_batch_put and [key for key in self._payload if key.startswith(k)]:
```

Comment:
Will this return keys that start with {k} also ? 
That is not expected here and we want only the exact match correct ?

Reply:
but self._consul.kv.get(k, recurse=True)
when recurse True it returns all the keys starting with the provided key

see here example:
```
from ha.util.consul_kv_store import  ConsulKvStore

consul = ConsulKvStore(prefix="cortx>ha>v2", host="consul-server.default.svc.cluster.local", port=8500)
print(f'{consul.get("events")}')
```

```
[root@cortx-ha-headless-svc util]# python3 test_kv.py
{'cortx>ha>v2>events>node>data>failed': '["hare"]', 'cortx>ha>v2>events>node>data>offline': '["hare"]', 'cortx>ha>v2>events>node>data>online': '["hare"]', 'cortx>ha>v2>events>node>server>failed': '["hare"]', 'cortx>ha>v2>events>node>server>offline': '["hare"]', 'cortx>ha>v2>events>node>server>online': '["hare"]', 'cortx>ha>v2>events>subscribe>hare': '["node>server>online", "node>server>offline", "node>server>failed", "node>data>online", "node>data>offline", "node>data>failed"]'}
```